services/verification/src/services/Injector.ts

```diff
@@ -356,6 +373,114 @@ export class Injector {
     };
   }
 
+  private async simulateCreationBytecode(
+    creationBytecode: string,
+    chainId: string,
+    evmVersion: string,
+    abiEncodedConstructorArguments?: string,
+    msgSender?: string
+  ): Promise<string> {
+    const stateManager = new DefaultStateManager();
+    const blockchain = await Blockchain.create();
+    const common = Common.custom({
+      chainId: parseInt(chainId),
+      defaultHardfork: evmVersion,
+    });
+    const eei = new EEI(stateManager, common, blockchain);
+
+    const evm = new EVM({
+      common,
+      eei,
+    });
+    if (creationBytecode.startsWith("0x")) {
```

Comment:
maybe we could start using some utility like with0x() or without0x() considering that this kind of check is so common. Not a priority btw, we can do it later

Reply:
Yes I'm doing the check non-sparingly. I wonder if there's a usual way to deal with it. I assume always having the 0x is the way to go like `sanitize0x(string)`

Reply:
In metamask they have two utils: with0x(), without0x()